### ISSUE TITLE
Issue:  how stream results with long context

### ISSUE THREAD

**Issue by qlql489** [langchain-ai/langchain#5532]
### Issue you'd like to raise.

i follow the chapter  “Chat Over Documents with Chat History” to build a bot chat with pdf，
i want to streanming return,
but when i use stuff chain like this
```python
    doc_chain = load_qa_chain(OpenAI(temperature=0), chain_type="stuff", prompt=QA_PROMPT)
    chain = ConversationalRetrievalChain(
        retriever=vector_db.as_retriever(),
        question_generator=question_generator,
        combine_docs_chain=doc_chain
    )
```

it return "This model's maximum context length is 4097 tokens, however you requested 5741 tokens (5485 in your prompt; 256 for the completion). Please reduce your prompt; or completion length"

when i use map_reduce chain

```python
doc_chain = load_qa_chain(OpenAI(temperature=0,streaming=True,callbacks=[StreamingStdOutCallbackHandler()]), chain_type="map_reduce", combine_prompt=getQaMap_reducePromot())
```
it  return "Cannot stream results with multiple prompts."

how to resolve it when the context is too long

**Comment by qixiang-mft** [comment]
you can streaming output result by  asyncio  AsyncIterator
```
        async def run_chain(prompt) -> AsyncIterator[str]:
            holder = AsyncIteratorCallbackHandler()

            async def run_task(p, callback):
                return await chain.arun(input=p, callbacks=[callback])

            task: Task = asyncio.get_running_loop().create_task(run_task(prompt, holder))

            full_text = ""
            async for text in holder.aiter():
                full_text = full_text + text
                yield full_text
            yield await task
```

**Comment by fitolobo** [comment]
Same problem here, I would like to stream the chain output, something like: 

`        llm =  OpenAI(streaming= True,
                    temperature = 0.0, 
                    model_name= "gpt-3.5-turbo",
                    callbacks=some_callback)

        chain = MultiRetrievalQAChain.from_retrievers(llm,
                                                    multi_retriever_info,
                                                    verbose=False)`
                                                    
to yield the tokens of the last answer. I could stream, and print the json intermediate step, however I couldn't get a generator following different instructions concerning the callbacks in the documentation.

**Comment by clemlesne** [comment]
Seems related to https://github.com/langchain-ai/langchain/issues/1349.

**Comment by straeter** [comment]
was this fixed in 2b3b4e0f600285d139d4e10a498530810d15b288 ? If so, can someone please create an example how to stream a load_qa_chain response?
